modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/rest/SW360ProjectClient.java

```diff
@@ -55,47 +60,48 @@ public String getEndpoint() {
 
         if (response.getStatusCode().is2xxSuccessful()) {
             SW360ProjectList resource = Optional.ofNullable(response.getBody())
-                    .orElseThrow(() -> new ExecutionException("Body was null"))
-                    .getContent();
-            if (resource != null &&
-                    resource.get_Embedded() != null &&
+                    .map(Resource::getContent)
+                    .orElseGet(() -> {
+                        LOGGER.debug("Request to search for project {} returned empty body", name);
+                        return new SW360ProjectList();
+                    });
+            if (resource.get_Embedded() != null &&
                     resource.get_Embedded().getProjects() != null) {
                 return resource.get_Embedded().getProjects();
             } else {
                 return new ArrayList<>();
             }
         } else {
-            throw new ExecutionException("Request to search for projects with the name " + name + " failed with "
-                    + response.getStatusCode());
+            LOGGER.debug("Request to search for projects with the name {} failed with {}",
+                    name, response.getStatusCode());
+            return new ArrayList<>();
         }
     }
 
     public SW360Project createProject(SW360Project sw360Project, HttpHeaders header) {
         HttpEntity<String> httpEntity = RestUtils.convertSW360ResourceToHttpEntity(sw360Project, header);
-        ResponseEntity<Resource<SW360Project>> response = doRestPOST(getEndpoint(), httpEntity,
-                new ParameterizedTypeReference<Resource<SW360Project>>() {});
-
-        if (response.getStatusCode() == HttpStatus.CREATED) {
-            return Optional.ofNullable(response.getBody())
-                    .orElseThrow(() -> new ExecutionException("Body was null"))
-                    .getContent();
-        } else {
-            throw new ExecutionException("Request to create project " + sw360Project.getName() + " failed with "
-                    + response.getStatusCode());
+        ResponseEntity<Resource<SW360Project>> response;
```

Comment:
There are many items of this pattern doSomeRequest -> checkResult, where the active code and the error handled are very much intermingled. Lets talk about a more clearer solution tomorrow, I would like to separate the error handling more from the business code.